### PR TITLE
fix #534 proper login routes.

### LINKED ISSUE #534: /login route is currently not working.
**What happened**:
In a frantic rush to meet deadlines the /login route went broke when moving it to the router. It should actually re-direct to the other sign-on page.

**What should have happened**:
Clicking the login button should re-direct it correctly. 

**How to reproduce it (as precise as possible)**: 
Click on the login button.

**Anything else we need to know?**: 
This will probably need some help understanding how the express router works at the moment, so we can discuss where the login information should actually be located.

**Environment**:

- OS: Windows 10 Pro, Windows 10 Home
- Browser: Chrome, Firefox

### PR DESCRIPTION
**Pull request by Grommers00**
#534 Issue This PR Addresses

1. Automatically close the issue when this PR is merged
    USAGE: Fixes #534

## Type of Change

- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

The login was done hurriedly last time. and wasn't properly tested, it now works again!

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
There is no need for tests currently, as it would need to test. 
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
No screenshots needed/
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)
Not applicable.

### CODE REVIEW

File: src/backend/web/routes/login.js [hunk 1]
```diff
@@ -3,8 +3,9 @@ const session = require('express-session');
 const bodyParser = require('body-parser');
 const cookieParser = require('cookie-parser');
 const passport = require('passport');
+require('../../login/usingPassport');
```
Comment (humphd) [inline src/backend/web/routes/login.js]: Why are we `require`'ing `../../login/usingPassport` twice? (line 6 and 7).  Can we consolidate this?

File: src/backend/web/routes/login.js [hunk 2]
```diff
@@ -21,13 +22,10 @@ router.use(passport.initialize({}));
...
-router.get('/login', handleLogin, passport.authenticate('samlStrategy'));
+router.get('/', handleLogin, passport.authenticate('samlStrategy'));
 
 /* this route is where you get directed from the external service. This is your redirect url found in the .env */
-router.post('/login/callback', handleLogin, passport.authenticate('samlStrategy'), function(
-  req,
-  res
-) {
+router.post('/callback', handleLogin, passport.authenticate('samlStrategy'), function(req, res) {
   logger.info({ user: req.user }, 'SSO login callback');
```
Comment (humphd) [inline src/backend/web/routes/login.js]: This seems like `debug` vs. `info` to me.

File: src/backend/web/routes/login.js [hunk 3]
```diff
@@ -21,13 +22,10 @@ router.use(passport.initialize({}));
...
   res.send('Log in Callback Success');
```
Comment (humphd) [inline src/backend/web/routes/login.js]: Necessary?  If needed, should this be text or JSON?  What does Seneca's SSO system do here, for comparison?

File: src/backend/web/routes/login.js (hunk 1 above)
Comment (Grommers00) [reply src/backend/web/routes/login.js]: Oops - was using this for testing purposes. Fixed.

File: src/backend/web/routes/login.js (hunk 2 above)
Comment (Grommers00) [reply src/backend/web/routes/login.js]: Switched to debug ideally this will be taken out shortly.

File: src/backend/web/routes/login.js (hunk 3 above)
Comment (Grommers00) [reply src/backend/web/routes/login.js]: I have it now returning a JSON object.

File: src/backend/web/routes/login.js [hunk 4]
```diff
@@ -21,24 +21,21 @@ router.use(passport.initialize({}));
...
 
 /* this route is where you get directed from the external service. This is your redirect url found in the .env */
-router.post('/login/callback', handleLogin, passport.authenticate('samlStrategy'), function(
-  req,
-  res
-) {
+router.post('/callback', handleLogin, passport.authenticate('samlStrategy'), function(req, res) {
   logger.info({ user: req.user }, 'SSO login callback');
-  res.send('Log in Callback Success');
+  res.send(JSON.stringify(req.user));
```
Comment (humphd) [inline src/backend/web/routes/login.js]: You can leave this an Object and have express deal with stringifying it:

```js
res.json(req.user);
```

**Pull request merged**